compose/project.py

```diff
@@ -592,8 +592,17 @@ def pull_service(service):
                 service.pull(ignore_pull_failures, silent=silent)
 
     def push(self, service_names=None, ignore_push_failures=False):
+        unique_images = []
```

Comment:
Not critical, but this would be better as a `set()`

Reply:
Yep!